### PR TITLE
Addapting OSS so rundeckpro can implement a new interface to save file on storage

### PR DESCRIPTION
**Pull request by carlosrfranco**
This PR is to create an interface so rundeckpro can implement it for a new feature to save file on storage.

### CODE REVIEW

File: core/src/main/java/com/dtolabs/rundeck/core/execution/ExecutionContext.java [hunk 1]
```diff
@@ -77,6 +78,14 @@
...
      */
     public StorageTree getStorageTree();
+    /**
+     * @return the file storage service
+     */
+    public StorageTree getFileStorageTree();
+    /**
+     * @return the file storage service
+     */
+    public AuthStorageTree getAuthStorageTree();
```
Comment (sjrd218) [inline core/src/main/java/com/dtolabs/rundeck/core/execution/ExecutionContext.java]: What is the difference between the file storage tree and the auth storage tree?

File: core/src/main/java/com/dtolabs/rundeck/core/logging/PluginLoggingContext.java [hunk 2]
```diff
@@ -51,4 +52,6 @@
      * @return the data context in the private scope
      */
     public DataContext getPrivateDataContext();
+
+    public StorageTree getFileStorateTree();
```
Comment (gschueler) [inline core/src/main/java/com/dtolabs/rundeck/core/logging/PluginLoggingContext.java]:
```suggestion
    public StorageTree getFileStorageTree();
```

**Comment by sjrd218** [comment]
@carlosrfranco I think you also need the changes to `application.yml` from the other PR. I don't think the app will bootstrap without it.

File: core/src/main/java/com/dtolabs/rundeck/core/storage/StorageTreeFactory.java [hunk 3]
```diff
@@ -138,6 +139,10 @@ public StorageTree createTree() throws Exception {
                     ), PathUtil.asPath("/keys")
             );
         }
+        if(null!=fileStorageConverterPluginAdapter && null!=defaultConverters && defaultConverters.contains("FileStorageLayer")){
+            logger.debug("Configuring base converter: FileStorageLayer" );
+            builder=builder.convert(fileStorageConverterPluginAdapter, PathUtil.asPath("/files"));
+        }
```
Comment (gschueler) [inline core/src/main/java/com/dtolabs/rundeck/core/storage/StorageTreeFactory.java]: the "defaultConverts" maybe should be refactored completely, allowing the factory resource definition to specify plugin instances/beans to use that way.  It was a small hack to have magic words "KeyStorageLayer", "StorageTimestamperConverter" that added certain converter plugins.  I think we should make a more generic way to do it

File: rundeckapp/grails-app/services/rundeck/services/AuthStorageManager.groovy [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.dtolabs.rundeck.core.authorization.AuthContext
+import com.dtolabs.rundeck.core.storage.ResourceMeta
+import com.dtolabs.rundeck.core.storage.StorageTree
+import com.dtolabs.rundeck.core.storage.StorageUtil
+import com.dtolabs.rundeck.core.storage.AuthStorageManager
+import grails.gorm.transactions.Transactional
+import org.rundeck.storage.data.MemoryTree
+
+@Transactional
+class AuthStorageTreeService implements AuthStorageManager{
```
Comment (gschueler) [inline rundeckapp/grails-app/services/rundeck/services/AuthStorageManager.groovy]: should avoid adding Service classes to the grails services, which don't need to be grails services.  This class could be removed, and simply inject the `rundeckAuthStorageManager` bean defined here https://github.com/rundeck/rundeck/pull/5392/files#diff-52259838c7d67dfe63914241dcf89443R114

Comment (gschueler) [reply rundeckapp/grails-app/services/rundeck/services/AuthStorageManager.groovy]: i think...

**Comment by gschueler** [comment]
closing as out of date

**Pull request closed without merging**